bitrise.yml

```diff
@@ -78,61 +69,58 @@ workflows:
     - restore-spm-cache@2.1.1:
         is_always_run: true
+    - xcode-build-for-test@3:
+        inputs:
+        - scheme: Fennec
+        - configuration: Fennec_Testing
+        - project_path: /Users/vagrant/git/firefox-ios/Client.xcodeproj
+        - xcodebuild_options: '"COMPILER_INDEX_STORE_ENABLE=NO" "CODE_SIGN_IDENTITY=" "CODE_SIGNING_REQUIRED=NO" "CODE_SIGNING_ALLOWED=NO"'
+        - destination: platform=iOS Simulator,name=iPhone 16,OS=18.2
     - script@1.2.1:
-        title: Build for Testing
+        title: Override xctestrun file to UnitTests
         inputs:
-        - content: |
+        - content: |-
+            #!/usr/bin/env bash
             set -euxo pipefail
-            echo "-- build-for-testing --"
-            mkdir DerivedData
 
-            xcodebuild -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile
-            xcodebuild "-project" "/Users/vagrant/git/firefox-ios/Client.xcodeproj" "-scheme" "Fennec" -configuration "Fennec_Testing" "CODE_SIGNING_ALLOWED=NO" -sdk "iphonesimulator" "-destination" "platform=iOS Simulator,name=iPhone 16,OS=18.2" "COMPILER_INDEX_STORE_ENABLE=NO" "build-for-testing" "CODE_SIGN_IDENTITY=" "CODE_SIGNING_REQUIRED=NO" "CODE_SIGNING_ALLOWED=NO" -derivedDataPath "/Users/vagrant/git/DerivedData" | xcpretty | tee xcodebuild_fennec.log
+            # Point to the Unit tests plan instead of Accessibility
+            UNIT_TESTS_XCTESTRUN="$BITRISE_TEST_BUNDLE_PATH/Fennec_UnitTest_iphonesimulator18.2-arm64.xctestrun"
+            SMOKE_UI_XCTESTRUN="$BITRISE_TEST_BUNDLE_PATH/Fennec_Smoketest_iphonesimulator18.2-arm64.xctestrun"
+
+            if [ ! -f "$UNIT_TESTS_XCTESTRUN" ]; then
+              echo "ERROR: .xctestrun file not found: $UNIT_TESTS_XCTESTRUN"
+              exit 1
+            fi
+
+            echo "Found UnitTest .xctestrun: $UNIT_TESTS_XCTESTRUN"
+            envman add --key BITRISE_XCTESTRUN_UNIT_TEST_FILE_PATH --value "$UNIT_TESTS_XCTESTRUN"
+            envman add --key BITRISE_XCTESTRUN_SMOKE_TEST_FILE_PATH --value "$SMOKE_UI_XCTESTRUN"
 
-            ls /Users/vagrant/git/DerivedData/Build/Products
-            ls /Users/vagrant/git/DerivedData
     - xcode-test-without-building@0.4.0:
-        run_if: '{{getenv "RUN_UI_TESTS" | eq "Run_UI_Tests"}}'
         timeout: 1200
         inputs:
         - destination: platform=iOS Simulator,name=iPhone 16,OS=18.2
         - xcodebuild_options: '"COMPILER_INDEX_STORE_ENABLE=NO" "CODE_SIGN_IDENTITY=" "CODE_SIGNING_REQUIRED=NO" "CODE_SIGNING_ALLOWED=NO"'
-        - xctestrun: "/Users/vagrant/git/DerivedData/Build/Products/Fennec_UnitTest_iphonesimulator18.2-arm64.xctestrun"
+        - xctestrun: $BITRISE_TEST_BUNDLE_PATH/Fennec_UnitTest_iphonesimulator18.2-arm64.xctestrun
         - maximum_test_repetitions: 2
-    - save-spm-cache@1.3.1:
+    - xcode-test-shard-calculation@0:
+        inputs:
+        - product_path: $BITRISE_XCTESTRUN_SMOKE_TEST_FILE_PATH
+        - shard_count: 6
```

Comment:
Was the `shard_count` determined by experimenting different number to optimize the wait time?